### PR TITLE
Np 876 refactor to nva commons part2

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -374,13 +374,14 @@ Resources:
...
     Properties:
+      CodeUri: create-upload
       Environment:
         Variables:
           S3_UPLOAD_BUCKET: !Ref  S3UploadBucket
           ALLOWED_ORIGIN:  '*'
       Handler: no.unit.nva.fileupload.CreateUploadHandler::handleRequest
       Runtime: java11
-      MemorySize: 1400
+      MemorySize: 1408
```
Comment (axthosarouris) [inline template.yaml:384]: why such number?

Comment (janbj) [reply template.yaml:384]: CodeGuru complained in an earlier PR, but on a second look 1408 is no better than 1400. I'll see if I can find the message again.

File: build.gradle [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
     id 'pmd'
-    id 'jacoco'
 }
 
 group 'no.unit.nva'
 version '1.0-SNAPSHOT'
 
-sourceCompatibility =  JavaVersion.VERSION_11  // source-code version and must be <= targetCompatibility
-targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
+
```
Comment (axthosarouris) [inline build.gradle:11]: since we are using java-11 features, I would recommend we keep the source and target compatibility

File: build.gradle [hunk 3]
```diff
@@ -18,82 +17,81 @@ allprojects {
...
+        failFast = true
+        finalizedBy jacocoTestReport
+    }
 
-pmd {
-    ruleSetConfig = rootProject.resources.text.fromFile('config/pmd/ruleset.xml')
-    ruleSets = []
-    ignoreFailures = false
-}
+    task buildZip(type: Zip) {
```
Comment (axthosarouris) [inline build.gradle:40]: is this a fat-jar?

File: build.gradle (hunk 2 above)
Comment (janbj) [reply build.gradle:11]: They are moved into subprojects {}

File: build.gradle (hunk 3 above)
Comment (janbj) [reply build.gradle:40]: No, and I don't know what this is used for. I will remove it.

**Pull request merged**